lint/cadence_v1_analyzer.go

```diff
@@ -0,0 +1,346 @@
+			case "getCapability":
+				v.newDiagnostic(
+					identifier,
+					"`getCapability` has been replaced by the Capability Controller API.",
+					"C1.0-CapabilityAPI-GetCapability",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#capability-controller-api-replaced-existing-linking-based-capability-api-82",
+				).WithSimpleReplacement("capabilities.get").Report()
+
+			// acct.getLinkTarget()
+			case "getLinkTarget":
+				v.newDiagnostic(
+					identifier,
+					"`getLinkTarget` has been replaced by the Capability Controller API.",
+					"C1.0-CapabilityAPI-GetLinkTarget",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#capability-controller-api-replaced-existing-linking-based-capability-api-82",
+				).Report()
+
+			// acct.addPublicKey()
+			case "addPublicKey":
+				v.newDiagnostic(
+					identifier,
+					"`addPublicKey` has been removed in favour of the new Key Management API. Please use `keys.add` instead.",
+					"C1.0-KeyAPI-AddPublicKey",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#capability-controller-api-replaced-existing-linking-based-capability-api-82",
+				).Report()
+
+			// acct.removePublicKey()
+			case "removePublicKey":
+				v.newDiagnostic(
+					identifier,
+					"`removePublicKey` has been removed in favour of the new Key Management API.\nPlease use `keys.revoke` instead.",
+					"C1.0-KeyAPI-RemovePublicKey",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#deprecated-key-management-api-got-removed-60",
+				).WithSimpleReplacement("keys.revoke").Report()
+			}
+		},
+	)
+}
+
+func (v *cadenceV1Analyzer) analyzeRemovedTypeIdentifiers() {
+	v.inspectTypeAnnotations(func(typeAnnotation *ast.TypeAnnotation) {
+		switch typeAnnotation.Type.String() {
+		case "AuthAccount":
+			v.
+				newDiagnostic(
+					typeAnnotation.Type,
+					"`AuthAccount` has been removed in Cadence 1.0.  Please use an authorized `&Account` reference with necessary entitlements instead.",
+					"C1.0-AuthAccount",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#account-access-got-improved-55",
+				).
+				WithSimpleReplacement("&Account").
+				Report()
+		case "PublicAccount":
+			v.
+				newDiagnostic(
+					typeAnnotation.Type,
+					"`PublicAccount` has been removed in Cadence 1.0.  Please use an `&Account` reference instead.",
+					"C1.0-PublicAccount",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#account-access-got-improved-55",
+				).
+				WithSimpleReplacement("&Account").
+				Report()
+		}
+	})
+}
+
+func (v *cadenceV1Analyzer) analyzeResourceDestructors() {
+	v.inspector.WithStack(
+		[]ast.Element{
+			(*ast.SpecialFunctionDeclaration)(nil),
+		},
+		func(element ast.Element, push bool, stack []ast.Element) bool {
+			declaration := element.(*ast.SpecialFunctionDeclaration)
+			if declaration.DeclarationIdentifier().Identifier != "destroy" {
+				return true
+			}
+
+			parent, ok := stack[len(stack)-2].(*ast.CompositeDeclaration)
```

Comment:
maybe do a size check before accessing `stack[len(stack)-2]`, to avoid any index out of bounds errors.